src/server/db/migrations/2020-11-13_0.3.0_to_0.4.0/migrate_0.3.0_to_0.4.0.js

```diff
@@ -0,0 +1,414 @@
+//    8. Get all the stories from the database
+//    9. Verify the update was successful
+
+// The command arguments are as follows:
+// node migrate_0.3.0_to_0.4.0.js $COUCHDBURL [maxDocsInBatchPerRequest]
+//
+// A sample command that runs this script on the "stories" database at "localhost:5984":
+// node migrate_0.3.0_to_0.4.0.js http://localhost:5984/stories
+//
+// @param {String} CouchDB-url - The url to the CouchDB where docoments should be migrated.
+// @param {Number} maxDocsInBatchPerRequest - [optional] Limit the number of stories to be processed in a batch.
+//     Default to 100 if not provided.
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    url = require("url");
+
+require("../utils/dbRequestUtils.js");
+
+var sjrk = fluid.registerNamespace("sjrk");
+fluid.registerNamespace("sjrk.storyUpdate");
+
+fluid.setLogging(fluid.logLevel.INFO);
+
+// Handle command line
+if (process.argv.length < 3) {
+    fluid.log("Usage: node migrate_0.3.0_to_0.4.0.js $COUCHDB_URL [maxDocsInBatchPerRequest]");
+    process.exit(1);
+}
+
+sjrk.storyUpdate.defaultMaxDocsInBatchPerRequest = 100;
+
+// these value should be updated with every migration
+sjrk.storyUpdate.oldSchemaVersion = undefined;
+sjrk.storyUpdate.newSchemaVersion = "0.4.0";
+
+sjrk.storyUpdate.uploadsDirectoryPrefix = "./uploads/";
+
+sjrk.storyUpdate.blockValuesToRemove = [
+    "imageUrl",                 // 3
+    "authoringEnabled",         // 4.a
+    "contentString",            // 4.b
+    "hasMobileCamera",          // 4.c
+    "hasTranscript",            // 4.d
+    "languageFromSelect",       // 4.e
+    "languageFromInput",        // 4.f
+    "savingEnabled",            // 4.g
+    "simplifiedText",           // 4.h
+    "transcript"                // 4.i
+];
+
+sjrk.storyUpdate.storyValuesToRemove = [
+    "authoringEnabled",         // 5.a
+    "categories",               // 5.b
+    "contentString",            // 5.c
+    "contentTypes",             // 5.d
+    "keywordString",            // 5.e
+    "languageFromSelect",       // 5.f
+    "languageFromInput",        // 5.g
+    "requestedTranslations",    // 5.h
+    "summary",                  // 5.i
+    "thumbnailAltText",         // 5.j
+    "thumbnailUrl",             // 5.k
+    "translationOf"             // 5.l
+];
+
+/**
+ * Create a set of options for data loader and a function to retreive them.
+ * The options are based on the command line parameters and a set of database
+ * constants
+ *
+ * @param {Array} processArgv - The command line arguments.
+ *
+ * @return {Object} - The options.
+ */
+sjrk.storyUpdate.initOptions = function (processArgv) {
+    var options = {};
+    options.couchDbUrl = processArgv[2];
+    options.maxDocsInBatchPerRequest = Number(processArgv[3]) || sjrk.storyUpdate.defaultMaxDocsInBatchPerRequest;
+
+    options.storiesUrl = options.couchDbUrl + "/_all_docs?include_docs=true&limit=" + options.maxDocsInBatchPerRequest;
```

Comment:
The script is not handling the `maxDocsInBatchPerRequest` properly by calling the recursive function `gpii.dbRequest.processRecursive()` to update documents in batch until all are updated.

Depending on number of documents you will update, as well as if there are other customers hosting their own SJRK servers that will use this migration script and their data volumes, you may either use your current logic but remove the support of `maxDocsInBatchPerRequest`, or add the recursive calls to properly support this feature.

Reply:
In the interest of saving time, I will opt for removing that feature for this iteration. If we run into issues with the batch size being too large, then we can revisit :)

Reply:
Sure. I have no objection.